### PR TITLE
KOGITO-5265: Quarkus DEV UI Integration: Form displayer

### PR DESCRIPTION
**Pull request by Sara4994**
Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting main: `[0.9.x] KOGITO-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Pull Request</b>  
  Please add comment: <b>Jenkins retest this</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

</details>

### CODE REVIEW

File: ui-packages/packages/form-details/webpack.dev.js [hunk 1]
```diff
@@ -39,13 +39,25 @@ module.exports = merge(common, {
...
           path.resolve(
             '../../node_modules/@patternfly/react-table/node_modules/@patternfly/react-styles/css'
+          ),
+          path.resolve(
+            '../../node_modules/@kogito-apps/form-displayer/dist/envelope/components/styles.css'
+          )
+        ],
+        use: ['style-loader', 'css-loader']
+      },
+      {
```
Comment (Sara4994) [inline ui-packages/packages/form-details/webpack.dev.js:49]: @AjayJagan you can add this include monaco related styles in the above, both looks same.

File: ui-packages/packages/form-details/webpack.prod.js [hunk 2]
```diff
@@ -47,6 +50,20 @@ module.exports = merge(common, {
             publicPath: '../../',
           }
         }, 'css-loader']
+      },
+      {
+        test: /\.css$/,
```
Comment (Sara4994) [inline ui-packages/packages/form-details/webpack.prod.js:55]: @AjayJagan same here.

File: ui-packages/packages/form-displayer/src/envelope/FormDisplayerEnvelopeView.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+ */
+
+import * as React from 'react';
+import { useImperativeHandle, useState } from 'react';
+import isEmpty from 'lodash/isEmpty';
+import { FormDisplayerChannelApi, FormArgs, FormInfo } from '../api';
+import { MessageBusClientApi } from '@kogito-tooling/envelope-bus/dist/api';
+import FormDisplayer from './components/FormDisplayer/FormDisplayer';
+
+import '@patternfly/patternfly/patternfly.css';
```
Comment (pefernan) [inline ui-packages/packages/form-displayer/src/envelope/FormDisplayerEnvelopeView.tsx]: This should be added in the ReactFormRenderer (or the loading), so the html renderer doesn't have the PF CSS loaded in the dom.

File: ui-packages/packages/form-displayer/package.json [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    "update-snapshot": "jest --updateSnapshot",
+    "test:coverage": "rimraf coverage && yarn test --coverage",
+    "lint": "eslint './src/**/*.ts{,x}'",
+    "format": "prettier --config ../../.prettierrc --check --write './src/**/*.{tsx,ts,js}'",
+    "clean": "rimraf dist"
+  },
+  "dependencies": {
+    "@kogito-apps/components-common": "^1.0.0",
+    "@kogito-apps/ouia-tools": "^1.0.0",
+    "@kogito-apps/task-console-shared": "^1.0.0",
```
Comment (pefernan) [inline ui-packages/packages/form-displayer/package.json]: I think we don't need this dep here

File: ui-packages/packages/form-displayer/src/envelope/components/ReactFormRenderer/ReactFormRenderer.tsx [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+  const [errorMessage, setErrorMessage] = useState<any>(null);
+
+  useEffect(() => {
+    /* istanbul ignore else */
+    if (source) {
+      renderform();
+    }
+  }, [source, resources]);
+
+  const renderform = () => {
```
Comment (pefernan) [inline ui-packages/packages/form-displayer/src/envelope/components/ReactFormRenderer/ReactFormRenderer.tsx:49]: I'd move this logic into a function in a separate module, just to keep the component clean.

File: ui-packages/packages/form-displayer/src/envelope/components/ReactFormRenderer/ReactFormRenderer.tsx [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+          formName,
+          trimmedSource
+        } = sourceHandler(source);
+
+        const scriptElement: HTMLScriptElement = document.createElement(
+          'script'
+        );
+
+        // @ts-ignore
+        window.PatternFly = window.PatternFlyReact;
```
Comment (pefernan) [inline ui-packages/packages/form-displayer/src/envelope/components/ReactFormRenderer/ReactFormRenderer.tsx:80]: I think we don't need this!

File: ui-packages/packages/form-displayer/webpack.dev.js [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+          path.resolve(
+            '../../node_modules/@patternfly/react-core/dist/esm/@patternfly/patternfly'
+          ),
+          path.resolve(
+            '../../node_modules/@patternfly/react-core/node_modules/@patternfly/react-styles/css'
+          ),
+          path.resolve(
+            '../../node_modules/@patternfly/react-table/node_modules/@patternfly/react-styles/css'
+          ),
+          path.resolve(
```
Comment (pefernan) [inline ui-packages/packages/form-displayer/webpack.dev.js]: I think we don't need this!

File: ui-packages/packages/form-displayer/webpack.dev.js [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+            '../../node_modules/@kogito-apps/task-console-shared/dist/envelope/styles.css'
+          )
+        ],
+        use: ['style-loader', 'css-loader']
+      }
+    ]
+  },
+  output: {
+    filename: 'envelope/components/styles.css'
```
Comment (pefernan) [inline ui-packages/packages/form-displayer/webpack.dev.js:65]: I'm worried about adding bundling the PF css into the component styles, this might be why we add the pf styles at the page top.

File: ui-packages/packages/runtime-tools-dev-ui-webapp/src/components/pages/FormDetailsPage/FormDetailsPage.tsx [hunk 9]
```diff
@@ -15,23 +15,42 @@
...
-
+  const history = useHistory();
+  const formData: FormInfo = history.location.state['formData'];
   return (
     <React.Fragment>
-      <PageSectionHeader titleText="Form Detail" />
+      <PageSection variant="light">
+        <PageTitle
+          title={formData.name}
+          extra={<Label variant="outline">{formData.type}</Label>}
```
Comment (pefernan) [inline ui-packages/packages/runtime-tools-dev-ui-webapp/src/components/pages/FormDetailsPage/FormDetailsPage.tsx:44]: here we show HTML/TSX, as UX suggested I'd change TSX for REACT, this is already done in the FormCard.tsx

**Review by pefernan — CHANGES_REQUESTED** [review]
@AjayJagan @Sara4994 good! all the functionalities are here, however there are some small changes to do:

- The PF CSS are loaded in the displayer even when the ReactFormRenderer isn't displayed. I'd say that this is a mix of how we bundle css into the styles.css and that the FormDisplayerEnvelopeView is importing the PF css. This isn not blocking but I'd take a look.
- In the ReactFormRenderer, if we write wrong react (break the jsx) and press play the syntax error appears (good!) if we fix the code and press play again.. the form-displayer appears blank. It needs a second click on the play button to finally show the form.

**Comment by pefernan** [comment]
@Sara4994 also it would be great if you rebase with master and take a look at the trusty package, there's a test failure there... PF renamed a TouchSpin component to NumberInput, try addressing this to have the build green.

**Comment by pefernan** [comment]
jenkins retest this please

**Comment by Sara4994** [comment]
Jenkins retest this

**Comment by Sara4994** [comment]
Jenkins retest this

**Review by AjayJagan — APPROVED** [review]
LGTM :) works fine and code looks well!!! Thanks Saravana

File: runtime-tools-quarkus-extension-parent/runtime-tools-quarkus-extension-deployment/pom.xml [hunk 10]
```diff
@@ -119,6 +119,25 @@
               </resources>
             </configuration>
           </execution>
+          <execution>
```
Comment (pefernan) [inline runtime-tools-quarkus-extension-parent/runtime-tools-quarkus-extension-deployment/pom.xml:122]: @paulovmr Since the **form-displayer** is going to be embedded into an iframe I had to add this config to copy the  .html & js nto the root folder of the _dev-static_  folder, so we can use the same _resources/form-displayer.html_ URL.
I think it's not optimal and the channel should provide the real path to the displayer, but that might require some changes that shouldn't be part of this PR.

**Comment by pefernan** [comment]
jenkins retest this please

**Comment by pefernan** [comment]
Code looks great, looking for the green...

File: runtime-tools-quarkus-extension-parent/runtime-tools-quarkus-extension-deployment/pom.xml (hunk 10 above)
Comment (paulovmr) [reply runtime-tools-quarkus-extension-parent/runtime-tools-quarkus-extension-deployment/pom.xml:122]: @pefernan I think it is ok to leave it like this in this PR, because it will change once we migrate to the monorepo.

**Review by paulovmr — APPROVED** [review]

**Comment by pefernan** [comment]
jenkins retest this please

**Comment by pefernan** [comment]
@kelvah here we are doing the version upgrade... do you want to give an eye?

**Comment by AjayJagan** [comment]
jenkins please retest this

**Comment by pefernan** [comment]
jenkins retest this

**Comment by pefernan** [comment]
ok, looks like jenkins cannot download node... last try

**Comment by pefernan** [comment]
jenkins retest this please

**Comment by kelvah** [comment]
> @kelvah here we are doing the version upgrade... do you want to give an eye?

Everything seems fine with Trusty. Thank you. 
I created [a JIRA](https://issues.redhat.com/browse/FAI-611) to track the extra toolbar spacing if we need to address it before the next PF update.

**Comment by pefernan** [comment]
@kelvah merged!
